Source/WebCore/html/parser/HTMLParserIdioms.cpp

```diff
@@ -522,4 +525,157 @@ std::optional<HTMLDimension> parseHTMLMultiLength(StringView multiLengthString)
+        Checked<unsigned, RecordOverflow> integer = 0;
+        while (rawInput.hasCharactersRemaining() && isASCIIDigit(*rawInput)) {
+            integer *= 10;
+            integer += (*rawInput - '0');
+            ++rawInput;
+        }
+
+        // The spec does not specify how to deal with arbitrarily large numbers, so we bail on overflow, falling back on "1*", matching the previous implementation.
+        // Filed https://github.com/whatwg/html/issues/11539 to track a standard solution.
+
+        if (integer.hasOverflowed()) [[unlikely]] {
+            result[i] = HTMLDimensionsListValue { .number = 1, .unit = HTMLDimensionsListValue::Unit::Relative };
+
+            skipUntil(rawInput, ',');
+            if (rawInput.hasCharactersRemaining())
+                ++rawInput;
+            continue;
+        }
+
+        value = integer.value();
+
+        // 5.7. If the character at position is U+002E (.), then:
+        if (rawInput.hasCharactersRemaining() && *rawInput == '.') {
+            ++rawInput;
+
+            // 5.7.1. Collect a sequence of code points consisting of ASCII whitespace and ASCII digits from input given position. Let `s` be the resulting sequence.
+            // 5.7.2. Remove all ASCII whitespace in `s`.
+
+            unsigned length = 0;
+            Checked<unsigned, RecordOverflow> fraction = 0;
+            while (rawInput.hasCharactersRemaining() && (isASCIIWhitespace(*rawInput) || isASCIIDigit(*rawInput))) {
+                if (isASCIIDigit(*rawInput)) {
+                    ++length;
+                    fraction *= 10;
+                    fraction += (*rawInput - '0');
+                }
+                ++rawInput;
+            }
+
+            // The spec does not specify how to deal with arbitrarily large numbers, so we bail on overflow, falling back on "1*", matching the previous implementation.
+            // Filed https://github.com/whatwg/html/issues/11539 to track a standard solution.
+
+            if (fraction.hasOverflowed()) [[unlikely]] {
+                result[i] = HTMLDimensionsListValue { .number = 1, .unit = HTMLDimensionsListValue::Unit::Relative };
+
+                skipUntil(rawInput, ',');
+                if (rawInput.hasCharactersRemaining())
+                    ++rawInput;
+                continue;
+            }
+
+            // 5.7.3. If `s` is not the empty string, then:
+            // 5.7.3.1. Let `length` be the number of characters in s (after the spaces were removed).
+            // 5.7.3.2. Let `fraction` be the result of interpreting s as a base-ten integer, and then dividing that number by 10^length.
+            // 5.7.3.3. Increment value by fraction.
+
+            if (length > 0)
+                value += (static_cast<double>(fraction) / std::pow(10, static_cast<double>(length)));
```

Comment:
Why the static_cast? I think this would do the exact same operation:
```suggestion
                value += fraction / std::pow(10.0, length);
```